### PR TITLE
Lars/information

### PR DESCRIPTION
**Pull request by Lars-Thestorf**
Add info function

Add function `--info` which will show information
about the text in the file.

Fix https://github.com/kinvolk/file-grinder/issues/4

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -75,6 +81,34 @@ func getOdd(data []Line) []Line {
...
+	for _, l := range data {
+		c += strings.Count(l.content, char)
+	}
+	return c
+}
+func getSymbolAmount(data []Line) int {
+	s := 0
+	for _, l := range data {
+		for _, c := range l.content {
+			if !(c == 32 || c == 9 || (c >= 48 && c <= 57) || (c >= 65 && c <= 90) || (c >= 97 && c <= 122)) {
```
Comment (alepuccetti) [inline main.go]: Add a comment to explain the if logic.

File: main.go [hunk 2]
```diff
@@ -75,6 +81,34 @@ func getOdd(data []Line) []Line {
...
+		}
+	}
+	return s
+}
+
+func printInfo(data []Line) {
+	fmt.Println()
+	fmt.Println("information:")
+	fmt.Println("-lines:\t\t\t", len(data))
+	fmt.Println("-characters:\t\t", getCharAmount(data, "")-len(data))
```
Comment (alepuccetti) [inline main.go]: Write a function called `getTotalCharacters(data)`, that returns the sum of the line lengths.

File: main.go [hunk 3]
```diff
@@ -75,6 +81,34 @@ func getOdd(data []Line) []Line {
...
+	fmt.Println("-white characters:\t", getCharAmount(data, " ")+getCharAmount(data, string(9)))
```
Comment (alepuccetti) [inline main.go]: Write a function called `getTotalWhiteCharacters(data)`, that returns the number of white characters.

File: main.go [hunk 4]
```diff
@@ -7,28 +7,34 @@ import (
...
 	lineNumbers := flag.Bool("show-line-numbers", false, "Show the linenumbers")
 	odd := flag.Bool("odd", false, "Show odd lines only")
 	even := flag.Bool("even", false, "Show even lines only")
+	info := flag.Bool("info", false, "Show Information about text in file")
 	flag.Parse()
 
 	data := readFile(*pathToFile)
-	if *even&&!*odd{
-		data=getEven(data)
+	if *even && !*odd {
```
Comment (LiliC) [inline main.go]: here something like a switch statment might make it more readable, just a suggestion... https://golang.org/doc/effective_go.html#switch

File: main.go [hunk 5]
```diff
@@ -81,18 +81,27 @@ func getOdd(data []Line) []Line {
 	return output
 }
 
-func getCharAmount(data []Line, char string) int {
+func getTotalCharacters(data []Line) int{
 	c := 0
 	for _, l := range data {
-		c += strings.Count(l.content, char)
+		c += strings.Count(l.content, "")-1
```
Comment (alepuccetti) [inline main.go]: please use `len(l.content)`, it is much more clear

File: main.go (hunk 4 above)
Comment (alepuccetti) [inline main.go]: I will open a new issue about it.

File: main.go (hunk 5 above)
Comment (Lars-Thestorf) [inline main.go]: if there is an unicode character like "Ö","Ä" or "Ü" it will count it as 2 characters

**Pull request merged**